grpc.go

```diff
@@ -1320,13 +1321,22 @@ func (a *agentGRPC) ReseedRandomDev(ctx context.Context, req *pb.ReseedRandomDev
 func (a *agentGRPC) GetGuestDetails(ctx context.Context, req *pb.GuestDetailsRequest) (*pb.GuestDetailsResponse, error) {
 	var details pb.GuestDetailsResponse
 	if req.MemBlockSize {
-		data, err := ioutil.ReadFile("/sys/devices/system/memory/block_size_bytes")
+		data, err := ioutil.ReadFile(sysfsMemoryBlockSizePath)
 		if err != nil {
-			return nil, err
-		}
-		details.MemBlockSizeBytes, err = strconv.ParseUint(string(data[:len(data)-1]), 16, 64)
-		if err != nil {
-			return nil, err
+			if os.IsNotExist(err) {
+				agentLog.WithField("sysfsMemoryBlockSizePath", sysfsMemoryBlockSizePath).Error("Guest kernel config doesn't support memory hotplug")
+				details.MemBlockSizeBytes = 0
+			} else {
+				return nil, err
+			}
+		} else {
+			if len(data) == 0 {
+				return nil, fmt.Errorf("/sys/devices/system/memory/block_size_bytes is empty")
```

Comment:
Could you use `sysfsMemoryBlockSizePath` in the error rather than hard-coding the path?

Reply:
ok, updated